### PR TITLE
Deauthorize deactivated user

### PR DESCRIPTION
**Pull request by eyouu**

### CODE REVIEW

**Review by vorfolomey — APPROVED** [review]

**Review by MarDatsko — APPROVED** [review]

**Review by Yura09 — APPROVED** [review]

File: core/src/main/java/greencity/security/filters/AccessTokenAuthenticationFilter.java [hunk 1]
```diff
@@ -24,35 +28,42 @@
...
         String token = jwtTool.getTokenFromHttpServletRequest(request);
+
         if (token != null) {
             try {
                 Authentication authentication = authenticationManager
                     .authenticate(new UsernamePasswordAuthenticationToken(token, null));
-                log.info("User successfully authenticate - {}", authentication.getPrincipal());
-                SecurityContextHolder.getContext().setAuthentication(authentication);
+                User user = userService.findByEmail((String) authentication.getPrincipal());
+                if (user.getUserStatus() != UserStatus.DEACTIVATED) {
```
Comment (antonbakalets) [inline core/src/main/java/greencity/security/filters/AccessTokenAuthenticationFilter.java:63]: My first idea was that this new if statement is not covered by unit test. Have you measured the test coverage for this class?

Comment (antonbakalets) [reply core/src/main/java/greencity/security/filters/AccessTokenAuthenticationFilter.java:63]: But then I've got a better idea. You don't have to load the user and then check the status. You should create a new method in `UserService` that will query the database for a user "by email and status equals not deactivated".

Comment (antonbakalets) [reply core/src/main/java/greencity/security/filters/AccessTokenAuthenticationFilter.java:63]: Another problem with the current approach is that if user is not found, a NPE will be thrown.

**Review by antonbakalets — CHANGES_REQUESTED** [review]

File: core/src/main/java/greencity/security/filters/AccessTokenAuthenticationFilter.java (hunk 1 above)
Comment (eyouu) [reply core/src/main/java/greencity/security/filters/AccessTokenAuthenticationFilter.java]: Done!

**Review by antonbakalets — APPROVED** [review]

**Pull request merged**